pkgs/os-specific/linux/systemd/vmlinux-btf.nix

```diff
@@ -0,0 +1,62 @@
+{
+  lib,
+  linuxManualConfig,
+  linuxConfig,
+  linuxHeaders,
+  pahole,
+  bpftools,
+  runCommand,
+}:
+let
+  # Create a minimal kernel config file, which is sufficient for the BTF
+  # that systemd uses.
+  configfile =
+    (linuxConfig {
+      inherit (linuxHeaders) src;
+      makeTarget = "allnoconfig";
+    }).overrideAttrs
+      (oldAttrs: {
+        depsBuildBuild = oldAttrs.depsBuildBuild ++ [ pahole ];
```

Comment:
Can we just add this to `linuxConfig`? It just seems like a footgun for custom config users otherwise. I wonder what other tools we're missing there...